.github/workflows/release.yaml

```diff
@@ -156,3 +156,96 @@ jobs:
           charts_dir: charts
           target_dir: charts
           linting: off
+
+  tagged-release-ghcr:
+    name: "Tagged Release GHCR"
+    runs-on: "ubuntu-22.04"
+    permissions:
+      packages: write
+      contents: read
+      actions: read
+    if: startsWith(github.ref, 'refs/tags/v') && github.repository == 'open-policy-agent/gatekeeper'
+    timeout-minutes: 30
+    steps:
+      - name: Harden Runner
+        uses: step-security/harden-runner@91182cccc01eb5e619899d80e4e971d6181294a7 # v2.10.1
+        with:
+          egress-policy: audit
+
+      - name: Check out code into the Go module directory
+        uses: actions/checkout@eef61447b9ff4aafe5dcd4e0bbf5d482be7e7871
+
+      - name: Set up Go
+        uses: actions/setup-go@0a12ed9d6a96ab950c8f026ed9f722fe0da7ef32 # v5.0.2
+        with:
+          go-version: "1.22"
+          check-latest: true
+
+      - name: Get tag
+        id: get-version
+        run: echo "TAG=${GITHUB_REF#refs/tags/}" >> $GITHUB_OUTPUT
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@c47758b77c9736f4b2ef4073d4d51994fabfe349 # v3.7.1
+
+      - name: Login to GitHub Container Registry
+        uses: docker/login-action@9780b0c442fbb1117ed29e0efdff1e18412f7567 # v3.3.0
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Check if ${{ env.IMAGE_REPO }} exists in GHCR
+        id: check-ghcr-image
+        run: |
+          if docker manifest inspect ghcr.io/${{ github.repository_owner }}/${{ env.IMAGE_REPO }}:${{ steps.get-version.outputs.TAG }} > /dev/null 2>&1; then
+            echo "exists=true" >> $GITHUB_OUTPUT
+          else
+            echo "exists=false" >> $GITHUB_OUTPUT
+          fi
+
+      - name: Build and push ghcr.io/${{ env.IMAGE_REPO }}
+        if: steps.check-ghcr-image.outputs.exists == 'false'
+        run: |
+          make REPOSITORY=ghcr.io/${{ env.IMAGE_REPO }} docker-buildx-release \
```

Comment:
can you integrate this into the existing release and pre-release flows so artifacts don't get rebuild and digests will be same

Reply:
Got it, I'll give it a try and update the PR.

Reply:
@sozercan I have made corrections based on the review comment. https://github.com/open-policy-agent/gatekeeper/pull/3658/commits/d1e8d7bf105cb8028d337a0e57e5fd6575d7d899